sciope/inference/rep_smc_abc.py

```diff
@@ -0,0 +1,187 @@
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Replenisment Sequential Monte-Carlo Approximate Bayesian Computation (SMC-ABC)
+"""
+
+# Imports
+from sciope.inference.abc_inference import ABC
+from sciope.inference.inference_base import InferenceBase
+from sciope.inference import abc_inference
+from sciope.core import core
+from sciope.utilities.distancefunctions import euclidean as euc
+from sciope.utilities.summarystats import burstiness as bs
+from sciope.utilities.housekeeping import sciope_logger as ml
+from sciope.utilities.priors.prior_base import PriorBase
+from sciope.utilities.epsilonselectors import RelativeEpsilonSelector
+
+import numpy as np
+import dask
+from dask.distributed import futures_of, as_completed, wait
+from dask import delayed
+
+class ReplenishmentSMCABC(InferenceBase):
+    """
+    Replenishment SMC - Approximate Bayesian Computation
+
+    Properties/variables:
+    * data                      (observed / fixed data)
+    * sim   					(simulator function handle)
+    * prior_function			(prior over the simulator parameters)
+    * perturbation_kernel       (kernel for perturbing parameters samples)
+    * summaries_function    	(summary statistics calculation function)
+    * distance_function         (function calculating deviation between simulated statistics and observed statistics)
+    * summaries_divisor         (numpy array of maxima - used for normalizing summary statistic values)
+    * use_logger    			(whether logging is enabled or disabled)
+
+    Methods:
+    * infer 					(perform parameter inference)
+    """
+
+    def __init__(self, data, sim, prior_function, perturbation_kernel,
+                 summaries_function=bs.Burstiness(),
+                 distance_function=euc.EuclideanDistance(),
+                 summaries_divisor=None, use_logger=False):
+
+        self.name = 'SMC-ABC'
+        super(ReplenishmentSMCABC, self).__init__(self.name, data, sim, use_logger)
+
+        self.prior_function = prior_function
+        self.summaries_function = summaries_function
+        self.distance_function = distance_function
+        self.summaries_divisor = summaries_divisor
+        self.perturbation_kernel = perturbation_kernel
+
+        if self.use_logger:
+            self.logger = ml.SciopeLogger().get_logger()
+            self.logger.info("Sequential Monte-Carlo Approximate Bayesian Computation initialized")
+
+    def _simulate_N(self, prior, N):
+
+        params_delayed = prior.draw(N)
+        params, = dask.compute(params_delayed, num_workers = 10)
```

Comment:
Remove `num_workers`, let user decide the amount of workers/processes/ threads by setting up Dask them self
```suggestion
        params, = dask.compute(params_delayed)
```